core/src/main/java/org/apache/iceberg/encryption/Ciphers.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.encryption;
+
+import java.security.GeneralSecurityException;
+import java.security.SecureRandom;
+import javax.crypto.AEADBadTagException;
+import javax.crypto.Cipher;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+public class Ciphers {
+  private static final int NONCE_LENGTH = 12;
+  private static final int GCM_TAG_LENGTH = 16;
+  private static final int GCM_TAG_LENGTH_BITS = 8 * GCM_TAG_LENGTH;
+
+  public static class AesGcmEncryptor {
+    private final SecretKeySpec aesKey;
+    private final Cipher cipher;
+    private final SecureRandom randomGenerator;
+
+    public AesGcmEncryptor(byte[] keyBytes) {
+      int keyLength = keyBytes.length;
+      Preconditions.checkArgument((keyLength == 16 || keyLength == 24 || keyLength == 32),
+          "Cannot use a key of length " + keyLength + " because AES only allows 16, 24 or 32 bytes");
+      this.aesKey = new SecretKeySpec(keyBytes, "AES");
+
+      try {
+        this.cipher = Cipher.getInstance("AES/GCM/NoPadding");
+      } catch (GeneralSecurityException e) {
+        throw new RuntimeException("Failed to create GCM cipher", e);
```

Comment:
Wonder if you want to define an Iceberg Crypto Runtime Exception just like you did for Parquet (ParquetCryptoRuntimeException). The reason is  RuntimeException is a very generic type and we will lose some meaning when converting from GeneralSecurityException.

Reply:
Mm, the callers won't have a type-specific reaction to this. The text itself provides the info for a post-mortem analysis. So probably this goes down to Iceberg's coding practices.